aws/offsite/lambda/verify-database-copy/index.js

```diff
@@ -0,0 +1,127 @@
+                        status_message = 'Terminating verification of database copy because database password change has not started yet.';
+                        console.error(status_message);
+                        postStatusToSlack(status_message);
+                        callback(new Error(status_message));
+                    } else {
+                        var connection;
+
+                        mysqlPromise.createConnection({
+                            host: DB_HOST,
+                            database: DB_NAME,
+                            user: DB_USER,
+                            password: DB_PASSWORD
+                        }).then(function (conn){
+                            connection = conn;
+                            return connection.query('SELECT count(*) AS number_of_users FROM users');
+                        }).then(function (rows){
+                            status_message = 'Successfully queried offsite backup of database.  Number of Users = ' + rows[0].number_of_users;
+                            console.log(status_message);
+                            postStatusToSlack(status_message);
+                            connection.end();
+
+                            var deleteDBInstanceParams = {
+                                DBInstanceIdentifier: DB_INSTANCE_IDENTIFIER,
+                                SkipFinalSnapshot: true
+                            };
+                            rds.deleteDBInstance(deleteDBInstanceParams, function (error, data) {
+                                if (error) {
+                                    console.log('Error deleting backup DB instance: ' + JSON.stringify(error), error.stack);
+                                } else {
+                                    console.log('Backup DB instance successfully deleted:  ' + JSON.stringify(data));
+                                }
+                            });
+                            callback(null, status_message);
+                        }).catch(function (error){
+                            if (connection && connection.end) {
+                                connection.end();
+                            }
+                            status_message = 'mysql error: ' + JSON.stringify(error);
+                            console.log(status_message);
+                            postStatusToSlack(status_message);
+                            callback(error);
+                        });
+                    }
+                }).catch(function (error){
+                    status_message = 'Terminating verification of database copy because database is not available due to pending password change.  ' + error.stack;
+                    console.error(status_message);
+                    postStatusToSlack(status_message);
+                    callback(error);
+                });
+            });
+        }).catch(function (error) {
```

Comment:
Nested `then` and `catch` are a bad sign. I think you already said your next step is to streamline these, but please ping me if you want help doing this.

Reply:
Thanks!  I don't fully understand how `reject` cascades down to the `catch`, but I'll try to sort that out this afternoon on my own.  I may need a few minutes of your time tomorrow if I get stuck.

Reply:
@davidsbailey I converted the nested `then` and `catch` to chained promises.  Can you take a look?